packages/core/src/connectors/types.ts

```diff
@@ -10,6 +15,7 @@ export interface ConnectorMetadata {
 }
 
 // The name `Connector` is used for database, use `ConnectorInstance` to avoid confusing.
+export type ConnectorInstanceWithConnector = ConnectorInstance & { connector: Connector };
```

Comment:
actually we don't need to separate connector instance "with connector" or "without". we can assume all `ConnectorInstance` s have the `.connector` prop:

```ts
type AllConnector = EmailConnector | SocialConnector;
type ConnectorInstance = AllConnector & { connector: Connector };
```